packages/p2p/src/components/my-profile/my-profile.jsx

```diff
@@ -1,70 +1,357 @@
+                    } else {
+                        setErrorMessage(response.error);
+                    }
+                }
+                resolve();
+            });
+        });
+    };
+
+    const getAdvertiserInfo = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_info: 1,
+            }).then(response => {
+                if (is_mounted.current) {
+                    if (!response.error) {
+                        const { p2p_advertiser_info } = response;
+                        setAdvertiserInfo(p2p_advertiser_info);
+                        setNickname(p2p_advertiser_info.name);
+                        setContactInfo(p2p_advertiser_info.contact_info);
+                        setDefaultAdvertDescription(p2p_advertiser_info.default_advert_description);
+                        setPaymentInfo(p2p_advertiser_info.payment_info);
+                    } else {
+                        setErrorMessage(response.error);
+                    }
+                    setIsLoading(false);
+                }
+                resolve();
+            });
+        });
+    };
+
+    const getAdvertiserStats = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_stats: 1,
+                id,
+            }).then(response => {
+                if (is_mounted.current) {
+                    if (!response.error) {
+                        const { p2p_advertiser_stats } = response;
+                        setStats(p2p_advertiser_stats);
+                    } else {
+                        setErrorMessage(response.error);
+                    }
+                }
+                resolve();
+            });
+        });
+    };
+
+    const handleSubmit = values => {
+        if (is_mounted.current) {
+            setIsButtonLoading(true);
+        }
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advertiser_update: 1,
+                contact_info: values.contact_info,
+                payment_info: values.payment_info,
+                default_advert_description: values.default_advert_description,
+            }).then(response => {
+                if (is_mounted.current) {
+                    if (!response.error) {
+                        const { p2p_advertiser_update } = response;
+                        setContactInfo(p2p_advertiser_update.contact_info);
+                        setDefaultAdvertDescription(p2p_advertiser_update.default_advert_description);
+                        setPaymentInfo(p2p_advertiser_update.payment_info);
+                        setIsSubmitSuccess(true);
+                    } else {
+                        setFormError(response.error);
+                    }
+                    setIsButtonLoading(false);
+                    setTimeout(() => setIsSubmitSuccess(false), 3000);
```

Comment:
`setIsSubmitSuccess` should be wrapped in a `is_mounted.current` check.

Reply:
It is... `if (is_mounted.current) {` ends at line 128, which is after this line.

Reply:
Done.